pkg/pillar/types/zedroutertypes.go

```diff
@@ -664,32 +670,125 @@ type NetworkInstanceConfig struct {
+	PortProbe NIPortProbe
+	// When EVE is deciding which port to use for multipath route and multiple ports have
+	// working connectivity (or probing is disabled), port can be selected based on the cost
+	// If this option is enabled, EVE will prefer ports with lower costs.
+	PreferLowerCost bool
+	// When EVE is deciding which port to use for multipath route and there are multiple
+	// candidates among cellular modems, it might make sense to consider the current
+	// cellular network signal strength. If this option is enabled, EVE will prefer
+	// cellular ports with better signal (only among cellular ports).
+	PreferStrongerWwanSignal bool
+}
+
+// ConnectivityProbeMethod -  method to use to determine the connectivity status of a port.
+type ConnectivityProbeMethod uint8
+
+const (
+	// ConnectivityProbeMethodNone : connectivity probing is disabled.
+	ConnectivityProbeMethodNone ConnectivityProbeMethod = iota
+	// ConnectivityProbeMethodICMP : use ICMP ping against the probed endpoint to determine
+	// the connectivity status.
+	ConnectivityProbeMethodICMP
+	// ConnectivityProbeMethodTCP : try to establish TCP connection with the probed endpoint
+	// to determine the connectivity status.
+	ConnectivityProbeMethodTCP
+)
+
+// ConnectivityProbe : configuration for user-defined connectivity-testing probe.
+type ConnectivityProbe struct {
+	// Method to use to determine the connectivity status.
+	Method ConnectivityProbeMethod
+	// ProbeHost is either IP or hostname.
+	ProbeHost string
+	// ProbePort is required for L4 probing methods (e.g. ConnectivityProbeMethodTCP).
+	ProbePort uint16
+}
+
+// String returns human-readable description of the probe.
+func (r ConnectivityProbe) String() string {
+	probeStr := "<none>"
+	switch r.Method {
+	case ConnectivityProbeMethodICMP:
+		probeStr = fmt.Sprintf("icmp://%s", r.ProbeHost)
+	case ConnectivityProbeMethodTCP:
+		probeStr = fmt.Sprintf("tcp://%s:%d", r.ProbeHost,
+			r.ProbePort)
+	}
+	return probeStr
+}
+
+// NIPortProbe is used to determine connectivity status of a port to decide if it is suitable
+// for the default route of a network instance.
+type NIPortProbe struct {
+	// EVE uses ICMP ping against the port's gateway IP to determine connectivity status.
+	// User can disable this probe method. This is typically needed when the gateway router
+	// is configured to drop/ignore ICMP pings and therefore this probe would return false
+	// negatives.
+	EnabledGwPing bool
+	// Ports exceeding this cost will have the gateway probing disabled to reduce
+	// traffic generated by probing (only less-frequent user-defined probe will be performed).
+	GwPingMaxCost uint8
+	// User-defined method to use to determine the port connectivity status.
+	// Zedrouter runs this additionally to gateway pings (unless EnabledGwPing is false).
+	UserDefinedProbe ConnectivityProbe
+}
+
+// String returns human-readable description of the route.
```

Comment:
Is this intended to be json syntax?
If so it might make sense to add a comment about that (and also mention that we can't use the standard json.Marshall since many fields are nore exported?).

Reply:
No, this is just made up format. I had to implement this String method otherwise using `fmt.Printf("%v", ...)` for `IPRouteConfig` would produce ugly output which was difficult to read in the logs.
Added comment that the format does not matter.